src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/ThunkedApis.cs

```diff
@@ -69,7 +69,19 @@ public sealed override string Location
             }
         }
 
-        public sealed override string CodeBase { get { throw new PlatformNotSupportedException(); } }
+        public sealed override string CodeBase
+        {
+            get
+            {
+                var assemblyPath = Location;
+                if (string.IsNullOrEmpty(assemblyPath))
+                {
+                    assemblyPath = Path.Combine(AppContext.BaseDirectory, typeof(object).Assembly.ManifestModule.Name);
+                }
+                return "file:///" + assemblyPath.Replace('\\', '/').TrimStart('/');
```

Comment:
Should this produce same result as CoreCLR?

https://github.com/dotnet/runtime/blob/0cf87c61d15bac2702d2d647434ca8a75cb9037a/src/coreclr/src/vm/pefile.cpp#L2153

Reply:
If i read the code correctly, it should produce the same, e.g. for unix, `CodeBase` looks like
```
file:///usr/share/dotnet/shared/Microsoft.NETCore.App/3.1.1/System.Private.CoreLib.dll
```
so it's always `file:///` + path with forward slashes (i trim a leading unix '/' from the path to ensure it's 3 and not 4 forward slashes).

Reply:
... except on network shares on windows, then it's 4 'file:////'...

Reply:
That seems to do it.